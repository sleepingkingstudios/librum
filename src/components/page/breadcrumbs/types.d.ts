export type Breadcrumb = {
  active?: true | false;
  label: string;
  url?: string;
};
