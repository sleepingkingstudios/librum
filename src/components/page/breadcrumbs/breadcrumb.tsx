import * as React from 'react';
import {
  Link,
  useLocation,
} from 'react-router-dom';

import type { Breadcrumb } from './types';

const EmptyBreadcrumb = ({ label }: { label: string }): JSX.Element => (
  <li className="page-breadcrumb">
    <span className="page-breadcrumb-empty">{ label }</span>
  </li>
);

const BreadcrumbLink = (
  { label, url }: { label: string, url: string }
): JSX.Element => (
  <li className="page-breadcrumb">
    <Link to={url} className="page-breadcrumb-link">{ label }</Link>
  </li>
)

export const PageBreadcrumb = (
  { breadcrumb }: { breadcrumb: Breadcrumb }
): JSX.Element => {
  const { active, label } = breadcrumb;
  const { pathname } = useLocation();

  if (!('url' in breadcrumb) || active === false) {
    return (<EmptyBreadcrumb label={label} />);
  }

  const { url } = breadcrumb;

  if (active !== true && pathname === url) {
    return (<EmptyBreadcrumb label={label} />);
  }

  return (<BreadcrumbLink label={label} url={url} />);
};
