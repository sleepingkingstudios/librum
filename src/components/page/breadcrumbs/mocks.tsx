import * as React from 'react';

import type { Breadcrumb } from './types';

const formatBreadcrumb = (breadcrumb: Breadcrumb): string => {
  const { active, label, url } = breadcrumb;

  return `${label}${url ? ` @ ${url}` : ''}${active ? ' (active)' : ''}`;
};

export const PageBreadcrumbs = ({
  breadcrumbs,
}: {
  breadcrumbs: Breadcrumb[],
}): JSX.Element => (
  <ul aria-label="breadcrumbs">
    {
      breadcrumbs.map((breadcrumb: Breadcrumb) => (
        <li aria-label="breadcrumb" key={breadcrumb.label}>
          { formatBreadcrumb(breadcrumb) }
        </li>
      ))
    }
  </ul>
);
