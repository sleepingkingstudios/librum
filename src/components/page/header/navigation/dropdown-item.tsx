import * as React from 'react';
import {
  Link,
  useLocation,
} from 'react-router-dom';

import { NavigationItemProps } from './item';
import { useThemeStyles } from '@themes';

export const DropdownItem = ({ label, url }: NavigationItemProps): JSX.Element => {
  const className = useThemeStyles('navigation-dropdown-item');
  const linkClassName = useThemeStyles('navigation-dropdown-link');
  const disabledClassName = useThemeStyles('navigation-dropdown-link-disabled');
  const { pathname } = useLocation();

  if (pathname === url) {
    return (
      <li className={className}>
        <p className={disabledClassName}>{ label }</p>
      </li>
    );
  } else {
    return (
      <li className={className}>
        <Link className={linkClassName} to={url}>
          { label }
        </Link>
      </li>
    );
  }
};
