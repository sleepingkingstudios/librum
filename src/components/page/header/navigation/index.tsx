import * as React from 'react';
import { map } from 'lodash';

import { joinClassNames } from '@utils/react-utils';
import {
  Dropdown,
  DropdownProps,
} from './dropdown';
import {
  NavigationItem,
  NavigationItemProps,
} from './item';
import { useThemeStyles } from '@themes';

export type NavigationProps = Array<NavigationItemProps | DropdownProps>;

interface PageNavigationProps {
  className?: string;
  navigation: NavigationProps;
}

const defaultListClassName = 'flex flex-col sm:flex-row';

export const PageNavigation = (
  { className, navigation }: PageNavigationProps
): JSX.Element => {
  const joinedClassNames = joinClassNames(
    'relative z-10',
    useThemeStyles('navigation'),
    className,
  );
  const joinedListNames = joinClassNames(
    defaultListClassName,
    useThemeStyles('navigation-list'),
  );
  const [open, setOpen] = React.useState<string | null>(null);

  return (
    <nav className={joinedClassNames}>
      <ul className={joinedListNames}>
        {
          map(
            navigation,
            (item: DropdownProps | NavigationItemProps): JSX.Element => {
              const { label } = item;

              if('items' in item) {
                return (
                  <Dropdown
                    key={label}
                    open={open}
                    setOpen={setOpen}
                    {...item}
                  />
                );
              }

              return (<NavigationItem key={label} {...item} />);
            },
          )
        }
      </ul>
    </nav>
  );
};
