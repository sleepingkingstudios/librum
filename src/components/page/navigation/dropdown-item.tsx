import * as React from 'react';
import {
  Link,
  useLocation,
} from 'react-router-dom';

import type { NavigationItemProps } from './item';

export const DropdownItem = ({ label, url }: NavigationItemProps): JSX.Element => {
  const { pathname } = useLocation();

  if (pathname === url) {
    return (
      <li className="dropdown-item">
        <p className="dropdown-item-link-disabled text-muted">{ label }</p>
      </li>
    );
  } else {
    return (
      <li className="dropdown-item">
        <Link className="dropdown-item-link" to={url}>
          { label }
        </Link>
      </li>
    );
  }
};
