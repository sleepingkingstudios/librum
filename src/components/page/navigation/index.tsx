import * as React from 'react';
import { map } from 'lodash';

import { joinClassNames } from '@utils/react-utils';
import {
  Dropdown,
  DropdownProps,
} from './dropdown';
import {
  NavigationItem,
  NavigationItemProps,
} from './item';

export type NavigationProps = Array<NavigationItemProps | DropdownProps>;

interface PageNavigationProps {
  className?: string;
  navigation: NavigationProps;
}

export const PageNavigation = (
  { className, navigation }: PageNavigationProps
): JSX.Element => {
  const joinedClassNames = joinClassNames(
    'page-navigation relative z-10',
    className,
  );
  const [open, setOpen] = React.useState<string | null>(null);

  return (
    <nav className={joinedClassNames}>
      <ul className="flex flex-col sm:flex-row">
        {
          map(
            navigation,
            (item: DropdownProps | NavigationItemProps): JSX.Element => {
              const { label } = item;

              if('items' in item) {
                return (
                  <Dropdown
                    key={label}
                    open={open}
                    setOpen={setOpen}
                    {...item}
                  />
                );
              }

              return (<NavigationItem key={label} {...item} />);
            },
          )
        }
      </ul>
    </nav>
  );
};
