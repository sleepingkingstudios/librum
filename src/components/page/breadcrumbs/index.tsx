import * as React from 'react';

import { PageBreadcrumb } from './breadcrumb';
import type { Breadcrumb } from './types';

export type { Breadcrumb } from './types';

export const PageBreadcrumbs = (
  { breadcrumbs }: { breadcrumbs: Breadcrumb[] }
): JSX.Element => {
  if (breadcrumbs.length === 0) { return null; }

  return (
    <ul className="flex flex-col sm:flex-row page-breadcrumbs">
      {
        breadcrumbs.map((breadcrumb: Breadcrumb): JSX.Element => {
          const { label } = breadcrumb;

          return (
            <PageBreadcrumb key={label} breadcrumb={breadcrumb} />
          );
        })
      }
    </ul>
  )
};
