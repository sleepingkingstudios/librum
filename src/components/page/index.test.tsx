import * as React from 'react';
import { v4 as generateUuid } from 'uuid';
import { faRadiation } from '@fortawesome/free-solid-svg-icons';

import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Page } from './index';
import { PageBreadcrumbs } from './breadcrumbs';
import type { Breadcrumb } from './breadcrumbs';
import { AlertsProvider } from '@alerts';
import type { Alert } from '@alerts';
import { actions } from '@session';
import type { User } from '@session';
import { render } from '@test-helpers/rendering';
import { createStore } from '@test-helpers/store';
import { PageNavigation } from './navigation';

describe('<Page>', () => {
  it('should render the header', () => {
    const defaultTitle = 'Librum';
    const defaultSubtitle = 'Campaign Companion';

    render(
      <Page>Page Content Here...</Page>,
      {
        router: true,
        store: true,
      }
    );

    expect(screen.getByText(defaultTitle)).toBeVisible();
    expect(screen.getByText(defaultSubtitle)).toBeVisible();
  });

  it('should render the footer', () => {
    const footerText = 'What lies beyond the furthest reaches of the sky?';

    render(
      <Page>Page Content Here...</Page>,
      {
        router: true,
        store: true,
      }
    );

    const footer = screen.getByText(footerText);

    expect(footer).toBeVisible();
  });

  it('should render the contents', () => {
    const content = 'Page Content Here...';

    render(
      <Page>Page Content Here...</Page>,
      {
        router: true,
        store: true,
      }
    );

    const contents = screen.getByText(content);

    expect(contents).toBeVisible();
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <Page>Page Content Here...</Page>,
      {
        router: true,
        store: true,
        theme: true,
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('with breadcrumbs: component', () => {
    const breadcrumbs: Breadcrumb[] = [
      {
        label: 'Breadcrumb One',
        url: '/',
      },
      {
        label: 'Breadcrumb Two'
      },
      {
        label: 'Breadcrumb Three',
        url: '/example',
      },
    ];
    const Breadcrumbs = () => (<PageBreadcrumbs breadcrumbs={breadcrumbs} />);

    it('should render the footer', () => {
      const footerText = 'What lies beyond the furthest reaches of the sky?';

      const { getByText } = render(
        <Page breadcrumbs={<Breadcrumbs />}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
        }
      );

      const footer = getByText(footerText);

      expect(footer).toBeVisible();

      breadcrumbs.forEach((breadcrumb: Breadcrumb) => {
        const { label } = breadcrumb;

        expect(getByText(label)).toBeVisible();
      });
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Page breadcrumbs={<Breadcrumbs />}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
          theme: true,
        },
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with breadcrumbs: value', () => {
    const breadcrumbs: Breadcrumb[] = [
      {
        label: 'Breadcrumb One',
        url: '/',
      },
      {
        label: 'Breadcrumb Two'
      },
      {
        label: 'Breadcrumb Three',
        url: '/example',
      },
    ];

    it('should render the footer', () => {
      const footerText = 'What lies beyond the furthest reaches of the sky?';

      const { getByText } = render(
        <Page breadcrumbs={breadcrumbs}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
        }
      );

      const footer = getByText(footerText);

      expect(footer).toBeVisible();

      breadcrumbs.forEach((breadcrumb: Breadcrumb) => {
        const { label } = breadcrumb;

        expect(getByText(label)).toBeVisible();
      });
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Page breadcrumbs={breadcrumbs}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
          theme: true,
        },
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with navigation: component', () => {
    const navigation = [
      {
        label: 'Home',
        url: '/',
      },
      {
        label: 'Launch Sites',
        url: '/launch-sites',
      },
      {
        label: 'Rockets',
        items: [
          {
            label: 'Engines',
            url: '/rockets/engines',
          },
          {
            label: 'Fuel Tanks',
            url: '/rockets/fuel-tanks',
          },
        ],
      },
      {
        label: 'Administration',
        items: [
          {
            label: 'Strategies',
            url: '/administration/strategies',
          },
        ],
      },
    ];
    const Navigation = () => (<PageNavigation navigation={navigation} />);

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Page navigation={<Navigation />}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
          theme: true,
        },
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with navigation: value', () => {
    const navigation = [
      {
        label: 'Home',
        url: '/',
      },
      {
        label: 'Launch Sites',
        url: '/launch-sites',
      },
      {
        label: 'Rockets',
        items: [
          {
            label: 'Engines',
            url: '/rockets/engines',
          },
          {
            label: 'Fuel Tanks',
            url: '/rockets/fuel-tanks',
          },
        ],
      },
      {
        label: 'Administration',
        items: [
          {
            label: 'Strategies',
            url: '/administration/strategies',
          },
        ],
      },
    ];

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Page navigation={navigation}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
          theme: true,
        },
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with subtitle: value', () => {
    it('should render the header', () => {
      const defaultTitle = 'Librum';
      const subtitle = 'Example Subtitle';

      render(
        <Page subtitle={subtitle}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
        },
      );

      expect(screen.getByText(defaultTitle)).toBeVisible();
      expect(screen.getByText(subtitle)).toBeVisible();
    });
  });

  describe('with title: value', () => {
    it('should render the header', () => {
      const title = 'Example Title';
      const defaultSubtitle = 'Campaign Companion';

      render(
        <Page title={title}>Page Content Here...</Page>,
        {
          router: true,
          store: true,
        },
      );

      expect(screen.getByText(title)).toBeVisible();
      expect(screen.getByText(defaultSubtitle)).toBeVisible();
    });
  });

  describe('when the session is authenticated', () => {
    const user: User = {
      email: 'alan.bradley@example.com',
      id: '00000000-0000-0000-0000-000000000000',
      role: 'user',
      slug: 'alan-bradley',
      username: 'Alan Bradley',
    };
    const token = '12345';
    const { create } = actions;

    it('should match the snapshot', () => {
      const { dispatch, store } = createStore();
      dispatch(create({ token, user }));

      const { asFragment } = render(
        <Page>Page Content Here...</Page>,
        {
          router: true,
          store,
          theme: true,
        },
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when there are many alerts', () => {
    const alerts: Alert[] = [
      {
        dismissable: true,
        message: 'Successfully activated reactor',
        persistent: false,
        type: 'success',
        uuid: generateUuid(),
      },
      {
        dismissable: false,
        icon: faRadiation,
        message: 'Reactor overheat imminent',
        persistent: false,
        type: 'warning',
        uuid: generateUuid(),
      },
      {
        dismissable: false,
        message: 'Increase in three-eyed fish observed',
        persistent: true,
        type: 'failure',
        uuid: generateUuid(),
      },
    ];

    it('should render the alerts', () => {
      const { getByText } = render(
        <AlertsProvider initialValue={alerts}>
          <Page>Page Content Here...</Page>
        </AlertsProvider>,
        {
          router: true,
          store: true,
        }
      );

      alerts.forEach((alert) => {
        const { message } = alert;
        const rendered = getByText(message);

        expect(rendered).toBeVisible();
      });
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <AlertsProvider initialValue={alerts}>
          <Page>Page Content Here...</Page>
        </AlertsProvider>,
        {
          router: true,
          store: true,
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
