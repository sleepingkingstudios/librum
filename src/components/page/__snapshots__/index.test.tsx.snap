// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page> should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 theme-default"
  >
    <header
      class="mb-3 text-center sm:text-left"
    >
      <span
        class="page-title"
      >
        Librum
      </span>
      <br
        class="block sm:hidden"
      />
      <small
        class="page-subtitle m-0 sm:ml-2"
      >
        Campaign Companion
      </small>
      <hr
        class="hr-fancy my-1"
      />
    </header>
    <main
      class="grow"
    >
      Page Content Here...
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left"
    >
      <hr
        class="hr-fancy mb-2"
      />
      <span
        class="text-muted"
      >
        What lies beyond the furthest reaches of the sky?
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<Page> when the session is authenticated should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 theme-default"
  >
    <header
      class="mb-3 text-center sm:text-left"
    >
      <span
        class="page-title"
      >
        Librum
      </span>
      <br
        class="block sm:hidden"
      />
      <small
        class="page-subtitle m-0 sm:ml-2"
      >
        Campaign Companion
      </small>
      <hr
        class="hr-fancy my-1"
      />
      <div
        class="text-left mt-2"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user mr-1"
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
            fill="currentColor"
          />
        </svg>
        <span
          class="hidden sm:inline"
        >
          You are currently logged in as 
        </span>
        <a
          href="/user"
        >
          Alan Bradley
        </a>
        <span
          class="float-right link-danger"
        >
          <button>
            Log Out
          </button>
        </span>
      </div>
    </header>
    <main
      class="grow"
    >
      Page Content Here...
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left"
    >
      <hr
        class="hr-fancy mb-2"
      />
      <span
        class="text-muted"
      >
        What lies beyond the furthest reaches of the sky?
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<Page> when there are many alerts should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 theme-default"
  >
    <header
      class="mb-3 text-center sm:text-left"
    >
      <span
        class="page-title"
      >
        Librum
      </span>
      <br
        class="block sm:hidden"
      />
      <small
        class="page-subtitle m-0 sm:ml-2"
      >
        Campaign Companion
      </small>
      <hr
        class="hr-fancy my-1"
      />
    </header>
    <main
      class="grow"
    >
      <ul
        class="alerts pb-3"
      >
        <li>
          <div
            class="alert alert-success"
          >
            <button
              class="float-right link-plain"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-x "
                data-icon="x"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-check pr-2"
              data-icon="circle-check"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
                fill="currentColor"
              />
            </svg>
            Successfully activated reactor
          </div>
        </li>
        <li>
          <div
            class="alert alert-warning"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-radiation pr-2"
              data-icon="radiation"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M215 186.7c-23.9 13.8-40 39.7-40 69.3L31 256c-17.7 0-32.2-14.4-30-31.9C9.7 154 46.8 92.7 100.3 52c14.1-10.7 33.8-5.3 42.7 10l72 124.7zM255 336c14.6 0 28.2-3.9 40-10.7l72 124.8c8.8 15.3 3.7 35.1-12.6 41.9c-30.6 12.9-64.2 20-99.4 20s-68.9-7.1-99.4-20c-16.3-6.9-21.4-26.6-12.6-41.9l72-124.8c11.8 6.8 25.4 10.7 40 10.7zm224-80l-144 0c0-29.6-16.1-55.5-40-69.3L367 62c8.8-15.3 28.6-20.7 42.7-10c53.6 40.7 90.6 102 99.4 172.1c2.2 17.5-12.4 31.9-30 31.9zM257 208a48 48 0 1 1 -2 96 48 48 0 1 1 2-96z"
                fill="currentColor"
              />
            </svg>
            Reactor overheat imminent
          </div>
        </li>
        <li>
          <div
            class="alert alert-failure"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-bomb pr-2"
              data-icon="bomb"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"
                fill="currentColor"
              />
            </svg>
            Increase in three-eyed fish observed
          </div>
        </li>
      </ul>
      Page Content Here...
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left"
    >
      <hr
        class="hr-fancy mb-2"
      />
      <span
        class="text-muted"
      >
        What lies beyond the furthest reaches of the sky?
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<Page> with breadcrumbs: component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 theme-default"
  >
    <header
      class="mb-3 text-center sm:text-left"
    >
      <span
        class="page-title"
      >
        Librum
      </span>
      <br
        class="block sm:hidden"
      />
      <small
        class="page-subtitle m-0 sm:ml-2"
      >
        Campaign Companion
      </small>
      <hr
        class="hr-fancy my-1"
      />
    </header>
    <main
      class="grow"
    >
      Page Content Here...
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left"
    >
      <ul
        class="flex flex-col sm:flex-row page-breadcrumbs"
      >
        <li
          class="page-breadcrumb"
        >
          <span
            class="page-breadcrumb-empty"
          >
            Breadcrumb One
          </span>
        </li>
        <li
          class="page-breadcrumb"
        >
          <span
            class="page-breadcrumb-empty"
          >
            Breadcrumb Two
          </span>
        </li>
        <li
          class="page-breadcrumb"
        >
          <a
            class="page-breadcrumb-link"
            href="/example"
          >
            Breadcrumb Three
          </a>
        </li>
      </ul>
      <hr
        class="hr-fancy mb-2"
      />
      <span
        class="text-muted"
      >
        What lies beyond the furthest reaches of the sky?
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<Page> with breadcrumbs: value should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 theme-default"
  >
    <header
      class="mb-3 text-center sm:text-left"
    >
      <span
        class="page-title"
      >
        Librum
      </span>
      <br
        class="block sm:hidden"
      />
      <small
        class="page-subtitle m-0 sm:ml-2"
      >
        Campaign Companion
      </small>
      <hr
        class="hr-fancy my-1"
      />
    </header>
    <main
      class="grow"
    >
      Page Content Here...
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left"
    >
      <ul
        class="flex flex-col sm:flex-row page-breadcrumbs"
      >
        <li
          class="page-breadcrumb"
        >
          <span
            class="page-breadcrumb-empty"
          >
            Breadcrumb One
          </span>
        </li>
        <li
          class="page-breadcrumb"
        >
          <span
            class="page-breadcrumb-empty"
          >
            Breadcrumb Two
          </span>
        </li>
        <li
          class="page-breadcrumb"
        >
          <a
            class="page-breadcrumb-link"
            href="/example"
          >
            Breadcrumb Three
          </a>
        </li>
      </ul>
      <hr
        class="hr-fancy mb-2"
      />
      <span
        class="text-muted"
      >
        What lies beyond the furthest reaches of the sky?
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<Page> with navigation: component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 theme-default"
  >
    <header
      class="mb-3 text-center sm:text-left"
    >
      <span
        class="page-title"
      >
        Librum
      </span>
      <br
        class="block sm:hidden"
      />
      <small
        class="page-subtitle m-0 sm:ml-2"
      >
        Campaign Companion
      </small>
      <nav
        class="page-navigation relative z-10"
      >
        <ul
          class="flex flex-col sm:flex-row"
        >
          <li
            class="navigation-item"
          >
            <p
              class="navigation-item-link-disabled text-muted"
            >
              Home
            </p>
          </li>
          <li
            class="navigation-item"
          >
            <a
              class="navigation-item-link"
              href="/launch-sites"
            >
              Launch Sites
            </a>
          </li>
          <li
            class="dropdown-container"
          >
            <div
              class="dropdown"
            >
              <button
                class="dropdown-label"
              >
                Rockets
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-caret-up fa-xs ml-2"
                  data-icon="caret-up"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="dropdown-container"
          >
            <div
              class="dropdown"
            >
              <button
                class="dropdown-label"
              >
                Administration
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-caret-up fa-xs ml-2"
                  data-icon="caret-up"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </li>
        </ul>
      </nav>
      <hr
        class="hr-fancy my-1"
      />
    </header>
    <main
      class="grow"
    >
      Page Content Here...
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left"
    >
      <hr
        class="hr-fancy mb-2"
      />
      <span
        class="text-muted"
      >
        What lies beyond the furthest reaches of the sky?
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<Page> with navigation: value should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 theme-default"
  >
    <header
      class="mb-3 text-center sm:text-left"
    >
      <span
        class="page-title"
      >
        Librum
      </span>
      <br
        class="block sm:hidden"
      />
      <small
        class="page-subtitle m-0 sm:ml-2"
      >
        Campaign Companion
      </small>
      <hr
        class="hr-fancy my-1 block sm:hidden"
      />
      <nav
        class="page-navigation relative z-10"
      >
        <ul
          class="flex flex-col sm:flex-row"
        >
          <li
            class="navigation-item"
          >
            <p
              class="navigation-item-link-disabled text-muted"
            >
              Home
            </p>
          </li>
          <li
            class="navigation-item"
          >
            <a
              class="navigation-item-link"
              href="/launch-sites"
            >
              Launch Sites
            </a>
          </li>
          <li
            class="dropdown-container"
          >
            <div
              class="dropdown"
            >
              <button
                class="dropdown-label"
              >
                Rockets
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-caret-up fa-xs ml-2"
                  data-icon="caret-up"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="dropdown-container"
          >
            <div
              class="dropdown"
            >
              <button
                class="dropdown-label"
              >
                Administration
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-caret-up fa-xs ml-2"
                  data-icon="caret-up"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </li>
        </ul>
      </nav>
      <hr
        class="hr-fancy my-1"
      />
    </header>
    <main
      class="grow"
    >
      Page Content Here...
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left"
    >
      <hr
        class="hr-fancy mb-2"
      />
      <span
        class="text-muted"
      >
        What lies beyond the furthest reaches of the sky?
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
